src/cookies.js

```diff
@@ -152,19 +153,15 @@ function checkOriginForSettingCookie(win, options, name) {
   if (options && options.allowOnProxyOrigin) {
     return;
   }
-  if (isProxyOrigin(win.location.href)) {
-    throw new Error(
-      'Should never attempt to set cookie on proxy origin: ' + name
-    );
-  }
-
+  devAssert(
```

Comment:
devAssert will be stripped in production, where the old code wouldn't. This has to be a user assert, or a some other non-eliminated throw.

Reply:
Wow, thanks for catching that. Big miss on my end.